test/test_utils/__init__.py

```diff
@@ -29,8 +29,26 @@
 DEFAULT_REGION = "us-west-2"
 # Constant to represent region where p3dn tests can be run
 P3DN_REGION = "us-east-1"
+def get_ami_id_boto3(region_name, ami_name_pattern):
+    """
+    For a given region and ami name pattern, return the latest ami-id
+    """
+    ami_list = boto3.client("ec2", region_name=region_name).describe_images(
+        Filters=[
+            {
+                "Name": "name",
+                "Values": [ami_name_pattern],
+            },
+        ],
+        Owners=[
+            'amazon',
+        ],
+    )
+    ami = sorted(ami_list["Images"], key=lambda x: x["CreationDate"], reverse=True)[0]
```

Comment:
Please use `max` instead. Applying sorting when finding one of the AMIs is alright, but I am sure this will be copy-pasted and applied to other the AMIs used for tests, in the future, and will increase the test startup time by a lot.
```suggestion
    ami = max(ami_list["Images"], key=lambda x: x["CreationDate"])
```
I think this suggestion should work, but I would recommend double-checking.

Reply:
Thank you for the review. I have checked it locally and it works perfectly fine.